solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -299,7 +361,8 @@ public long setSizeWriter(LongObjectProcedure<String> sizeWriter) {
     @SuppressWarnings("try")
     public void close() throws IOException {
       try (backing) {
-        fileSizeMap.put(name, backing.getFilePointer());
+        // TODO: get onDiskSize correctly here
+        fileSizeMap.put(name, new Sizes(backing.getFilePointer(), backing.getFilePointer()));
```

Comment:
@magibney If I'm reading the code correctly, this should be accurate for `CompressingDirectory`, since it's compressing as it goes?